### PR TITLE
[Beats] Build darwin/arm64 and universal binary

### LINKED ISSUE #29583: [Elastic Agent] Elastic Agent to compile natively on a Mac M1 machine
**Describe the enhancement:**
We need Elastic Agent to be compiled so it can execute on a Mac with the M1 chip without any emulation software. With MacOS 11 the lowest required operating system.

**Describe a specific use case for the enhancement or feature:**
We need the Elastic Agent to be compiled so it can execute on a Mac with the M1 chip without any emulation software. With MacOS 11 the lowest required operating system.

Replaces: #26683
Relates: #26396

### PR DESCRIPTION
**Pull request by AndersonQ**
## What does this PR do?

It enables the Elastic-Agent and Beats to be built for `darwin/arm64` and generates a universal binary (`darwin/universal`).

The [universal binary](https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary) is the merge of the `darwin/arm64` with `darwin/amd64`. Therefore it requires to first build the  `darwin/arm64` and `darwin/amd64` and them _"merge"_ them together into the `darwin/universal`. Also `darwin/universal` isn't a valid platform or OS/ARCH pair, therefore it isn't recognised by the Go compiler. Given that, the choice was to add `darwin/arm64` as a new platform for the Elastic-Agent and when ever both platforms (`darwin/arm64` and `darwin/amd64`) are built, also generate and package the universal binary.
Even though  it's ~quite~ an implicit behaviour, the alternative would also require a lot of changes and exceptions to have our build process recognising `darwin/universal`, but compiling `darwin/arm64` and `darwin/amd64` and merging them. Also the current process makes available all three of them, allowing the user to choose which fits better their needs. The drawback of the universal binary is its size, the double of the single binaries.

A new target, `BuildDarwinUniversal`  was added on the `x-pack`s `magefile`s required by the Elastic-Agent.

Also fixes some small typos, redundant information on the help command and remove the use of some deprecated functions.

## Why is it important?

We need to support Mac with M1 chips

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [x] build darwin/arm64 binary
- [x] build universal 2 binary
- [x] use mage to automatically build the universal binary

## How to test this PR locally
1. Build the elastic-agent for the new platforms:
   - **darwin/arm64**: `PLATFORMS="darwin/arm64" PACKAGES="tar.gz" mage package`
   - **darwin/universal**: `PLATFORMS="darwin/arm64,darwin/amd64" PACKAGES="tar.gz" mage package`

2. Run the elastic-agent

####**Watch out:**
The elastic-agent tries to download any binary it might need to comply with the assigned policy. However none of the new binaries from this PR are present on the artifact API and other applications (e.g. fleet-server and elastic-endpoint) also aren't publishing `darwin/arm64` versions. Therefore if the elastic-agent tries to download them, it'll fail. This is the expected behavior and this PR does not address it, see *Out of Scope*.

## Out of scope:
 - Handle downloading binaries that does not publish `darwin-universal` artifacts on the artifact API
 - Handle upgrading a `darwin/amd64` agent running under Rosetta2 on a Mac with M1 chip. This will be addressed on another PR.

## Related issues

- Relates elastic/elastic-agent#151
- Closes elastic/beats#29582

### CODE REVIEW

**Comment by v1v** [comment]
This feature will be enabled no just in ElasticAgent but in all the other Beats, is that correct?

**Comment by ph** [comment]
@AndersonQ I've tried to build all the artifacts with `make release` in the top level directory and it fails on my side 

```
>> Building using: cmd='build/mage-linux-amd64 buildDarwinUniversal', env=[CC=oa64-clang, CXX=oa64-clang++, GOARCH=arm64, GOARM=, GOOS=darwin, PLATFORM_ID=darwin-arm64]
Unknown target specified: "buildDarwinUniversal"
Error: failed building for darwin/arm64: exit status 2
failed building for darwin/arm64: exit status 2
package ran for 14m30.180628207s
Error: failed merging darwin/amd64 and darwin/arm64 into darwin/universal target=buildDarwinUniversal for platform=darwin/arm64: running "docker run --env EXEC_UID=501 --env EXEC_GID=20 -v /Users/ph/go/pkg/mod:/go/pkg/mod:ro --rm --env GOFLAGS=-mod=readonly --env MAGEFILE_VERBOSE= --env MAGEFILE_TIMEOUT= --env SNAPSHOT=false --env DEV=false -v /Users/ph/src/beats:/go/src/github.com/elastic/beats -w /go/src/github.com/elastic/beats/auditbeat docker.elastic.co/beats-dev/golang-crossbuild:1.17.5-darwin-arm64-debian10 --build-cmd build/mage-linux-amd64 buildDarwinUniversal -p darwin/arm64" failed with exit code 1
failed merging darwin/amd64 and darwin/arm64 into darwin/universal target=buildDarwinUniversal for platform=darwin/arm64: running "docker run --env EXEC_UID=501 --env EXEC_GID=20 -v /Users/ph/go/pkg/mod:/go/pkg/mod:ro --rm --env GOFLAGS=-mod=readonly --env MAGEFILE_VERBOSE= --env MAGEFILE_TIMEOUT= --env SNAPSHOT=false --env DEV=false -v /Users/ph/src/beats:/go/src/github.com/elastic/beats -w /go/src/github.com/elastic/beats/auditbeat docker.elastic.co/beats-dev/golang-crossbuild:1.17.5-darwin-arm64-debian10 --build-cmd build/mage-linux-amd64 buildDarwinUniversal -p darwin/arm64" failed with exit code 1
make[1]: *** [release] Error 1
make: *** [release] Error 1
```

I also try to use the `mage buildDarwinUniversal` in the `x-pack/elastic-agent` without luck? 
Am I missing something ?

**Review by ph — CHANGES_REQUESTED** [review]
Should have used request changes before for my last comment.

**Comment by AndersonQ** [comment]
@v1v 
> This feature will be enabled no just in ElasticAgent but in all the other Beats, is that correct?

Perhaps, I'll double check with the data plane

**Comment by AndersonQ** [comment]
> @AndersonQ I've tried to build all the artifacts with `make release` in the top level directory and it fails on my side

That's probably because not all beats have the `buildDarwinUniversal` target. I'll check and add the missing ones

> I also try to use the `mage buildDarwinUniversal` in the `x-pack/elastic-agent` without luck? Am I missing something ?

Probably, that's the catch of the approach I took. `buildDarwinUniversal` assumes that the `darwin/amd64` and `darwin/arm64`  were built as this rule only merges them 2 into the universal binary. Perhaps another name instead of _build..._ would be better. 
I'm open to suggestions... 
 - _createDarwinUniversal_, 
 - _mergeDarwinUniversal_ (I don't like _merge_, seems that `DarwinUniversal` will be merged into something), 
 - _assembleDarwinUniversal_ (I think this is the best so far...)

**Comment by ph** [comment]
I think `assemble` is also more appropriate, I was using `make release` because this is what the release manager will do, and Agent and his subprocess Filebeat, metricbeat, etc need to be compiled as universal.

File: heartbeat/magefile.go
Comment (andrewkroh) [inline heartbeat/magefile.go:93]: This looks like it could be added to `github.com/elastic/beats/v7/dev-tools/mage/target/build`. If so, then any of the Beats that are directly using `mage:import` of that package won't need to duplicate it.

File: dev-tools/mage/crossbuild.go
Comment (andrewkroh) [inline dev-tools/mage/crossbuild.go:421]: It also looks like it assumes that `lipo` is present in your `PATH`. Does this imply that this can be only be run macos machines with an xcode toolchain installed?

If so it would be good to document this assumption and possibly add an [`LookPath`](https://pkg.go.dev/os/exec#LookPath) check that provides a descriptive error about the requirements and how to get the command.

Comment (andrewkroh) [reply dev-tools/mage/crossbuild.go:421]: IIUC this is automatically invoked by the `CrossBuild` target when darmin/amd64 and darmin/arm64 are platform targets. And it gets execute inside the crossbuilder container. If that's correct that would be good to mention in godoc.

File: x-pack/elastic-agent/magefile.go
Comment (andrewkroh) [inline x-pack/elastic-agent/magefile.go:317]: So does the Agent binary not actually include any of the individual Beat universal binaries? IIUC this is still packing an arm64 and amd64 Beat package within the Agent universal package.

If that's correct then who is intended consumer of the universal packages for the Beats. Will those be made available for download by end-users as part of the official releases?

Comment (AndersonQ) [reply x-pack/elastic-agent/magefile.go:318]: The beats are build for `darwin/arm64` as well as `darwin/universal`. 
Whenever both `darwin/amd64` and `darwin/arm64` are built, a `darwin/universal` is also built and packaged. 
The final `elastic-agent tar.gz` will contain the universal agent and the beats for both `darwin/amd64` and `darwin/arm64`. Thus the agent will install the platform specific beat.

**Comment by aleksmaus** [comment]
👍 nice! what's the ETA on this getting merged? no pressure, just curious. could use this.

**Comment by aleksmaus** [comment]
tried this branch on MacBook Pro M1, Mac OS 12.2, Docker Desktop 4.4.2 (73305) 
got segfault:
```
Status: Downloaded newer image for docker.elastic.co/beats-dev/golang-crossbuild:1.17.6-darwin-arm64-debian10
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
>> Building using: cmd='build/mage-linux-arm64 golangCrossBuild', env=[CC=oa64-clang, CXX=oa64-clang++, GOARCH=arm64, GOARM=, GOOS=darwin, PLATFORM_ID=darwin-arm64]
>> build: Building filebeat-darwin-arm64
Running target: GolangCrossBuild
exec: git rev-parse HEAD
Adding build environment vars: map[CGO_ENABLED:1]
exec: go build -o build/golang-crossbuild/filebeat-darwin-arm64 -buildmode pie -gcflags "all=-N -l" -ldflags -X github.com/elastic/beats/v7/libbeat/version.buildTime=2022-02-07T23:10:38Z -X github.com/elastic/beats/v7/libbeat/version.commit=0185ca18e014523082fe0d4f0c5ed3ae0acf76fb
# runtime/cgo
qemu: uncaught target signal 11 (Segmentation fault) - core dumped
```

**Comment by AndersonQ** [comment]
> 



> tried this branch on MacBook Pro M1, Mac OS 12.2, Docker Desktop 4.4.2 (73305) got segfault:
> 
> ```
> Status: Downloaded newer image for docker.elastic.co/beats-dev/golang-crossbuild:1.17.6-darwin-arm64-debian10
> WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
> >> Building using: cmd='build/mage-linux-arm64 golangCrossBuild', env=[CC=oa64-clang, CXX=oa64-clang++, GOARCH=arm64, GOARM=, GOOS=darwin, PLATFORM_ID=darwin-arm64]
> >> build: Building filebeat-darwin-arm64
> Running target: GolangCrossBuild
> exec: git rev-parse HEAD
> Adding build environment vars: map[CGO_ENABLED:1]
> exec: go build -o build/golang-crossbuild/filebeat-darwin-arm64 -buildmode pie -gcflags "all=-N -l" -ldflags -X github.com/elastic/beats/v7/libbeat/version.buildTime=2022-02-07T23:10:38Z -X github.com/elastic/beats/v7/libbeat/version.commit=0185ca18e014523082fe0d4f0c5ed3ae0acf76fb
> # runtime/cgo
> qemu: uncaught target signal 11 (Segmentation fault) - core dumped
> ```

It seems to be a M1<->docker problem, see [for-mac#5123](https://github.com/docker/for-mac/issues/5123)

Regarding the ETA to merge, well the original plan was 8.1, but it might be postponed to 8.2. Besides that, I don't have an ETA to merge, just as soon as it's ready :)

**Comment by aleksmaus** [comment]
> Interesting, well it's with the robots, they provide the images and the support for crossbuild.

Not sure if this helps, opened the issue:
https://github.com/elastic/golang-crossbuild/issues/153
Maybe will get clarification if there is a problem with the builder image.

**Comment by amolnater-qasource** [comment]
Hi @AndersonQ 
We have revalidated this on latest 8.1 Snapshot with all the provided artifacts.

**We had below observations:**
<HTML>

<BODY TEXT="#000000">
<TABLE FRAME=VOID CELLSPACING=0 COLS=3 RULES=NONE BORDER=0>
	<COLGROUP><COL WIDTH=295><COL WIDTH=86><COL WIDTH=86></COLGROUP>
	<TBODY>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" WIDTH=295 HEIGHT=66 ALIGN=CENTER VALIGN=MIDDLE><B>Artifact Name</B></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" WIDTH=86 ALIGN=CENTER VALIGN=MIDDLE><B>Machine</B></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" WIDTH=86 ALIGN=CENTER VALIGN=MIDDLE><B>Status on adding Endpoint Security</B></TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=34 ALIGN=CENTER VALIGN=MIDDLE>elastic-agent+endpoint-8.1.0-SNAPSHOT-darwin-x86_64.tar</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Monterey 12</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Healthy</TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=34 ALIGN=CENTER VALIGN=MIDDLE> elastic-agent+endpoint-8.1.0-SNAPSHOT-darwin-universal.tar</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Monterey 12</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Healthy</TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=34 ALIGN=CENTER VALIGN=MIDDLE>only elastic-agent-8.1.0-SNAPSHOT-darwin-x86_64.tar</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Monterey 12</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Healthy</TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=34 ALIGN=CENTER VALIGN=MIDDLE>only elastic-agent-8.1.0-SNAPSHOT-darwin-universal.tar</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Monterey 12</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Healthy</TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=34 ALIGN=CENTER VALIGN=MIDDLE>only elastic-agent-8.1.0-SNAPSHOT-darwin-arm64.tar</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Mac M1</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Unhealthy(logs attached)</TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=34 ALIGN=CENTER VALIGN=MIDDLE>elastic-agent+endpoint-8.1.0-SNAPSHOT-darwin-arm64.tar</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Mac M1</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Unhealthy(logs attached)</TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=34 ALIGN=CENTER VALIGN=MIDDLE>elastic-agent+endpoint-8.1.0-SNAPSHOT-darwin-universal.tar</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Mac M1</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Unhealthy(logs attached)</TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=34 ALIGN=CENTER VALIGN=MIDDLE>only elastic-agent-8.1.0-SNAPSHOT-darwin-universal.tar</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Mac M1</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>Unhealthy(logs attached)</TD>
		</TR>
	</TBODY>
</TABLE>
</BODY>

</HTML>

**Build details:**
BUILD: 50363
COMMIT: 05408fe74aee0f7161fe4b1568b71e91fb030874

**Logs on adding Endpoint to M1 machine:**
[elastic-agent-universal.zip](https://github.com/elastic/beats/files/8088428/elastic-agent-universal.zip)

Please let us know if anything else is required from our end.
Thanks

**Comment by AndersonQ** [comment]
@amolnater-qasource this zip file seems to be empty. there is a `log` folder inside it, but that is all, the folder is empty.

**Comment by AndersonQ** [comment]
@amolnater-qasource another thing, how did you get this logs? I believe it'd be better if you'd use the diagnostics command to gather not only the logs, but pretty much everything: policy, the beats logs...

File: .ci/packaging.groovy
Comment (v1v) [inline .ci/packaging.groovy:171]: In order to test these changes you need to use the `UI`:
- Go to https://beats-ci.elastic.co/job/Beats/job/packaging/view/change-requests/job/PR-29585/build?delay=0sec and select `macos`

Although I don't know what's the current status, since this particular stage is not executed often.

Comment (v1v) [reply .ci/packaging.groovy:171]: Actually, It won't work -> https://beats-ci.elastic.co/label/macosx-10.12/ is not anymore available, so you need to change the label to use a different one, you can use https://beats-ci.elastic.co/label/macosx/, 

change https://github.com/elastic/beats/pull/29585/files#diff-5edb5ab93fc95960129eecb2ee3d8763a9c5e02b56e5b80126e0c5a52296ee8fR158 from `agent { label 'macosx-10.12' }` with `agent { label 'macosx' }`

Though I cannot warranty what's the status for those macos workers, as they were corrupted sometime ago, so, if any issues then you might need to contact the infra team as they provision and manage those macos workers.

**Comment by v1v** [comment]
I've got some questions:
1) Is this particular universal binary something to be released as part of the Unified Release process?
2) Is it required to use `macosx` with `docker?

Then, it's worth to involve the CI System and Release teams as they are the ones who provide:
- MacOS workers in the CI
- Configure docker in those workers
- Run the release process as part of the unified release process.

On the other hand, I tried to solve -> https://github.com/elastic/beats/pull/29585#discussion_r811228476 in https://github.com/elastic/beats/pull/30505 but I'm now facing issues with the docker-machine not being available at runtime. But before doing anything else, we might need to get some clarity about the above questions. `MacOSX` workers are not bullet proof

**Comment by amolnater-qasource** [comment]
Hi @AndersonQ 
There was some issue in copying the logs.
- We have revalidated adding endpoint security on M1 universal artifact agent.
- Agent directly goes to unhealthy state when endpoint security is added.

Please find below logs for the same using diagnostics command :
[elastic-agent-diagnostics-2022-02-24T04-51-13Z-00.zip](https://github.com/elastic/beats/files/8129747/elastic-agent-diagnostics-2022-02-24T04-51-13Z-00.zip)

Please let us know if anything else is required from our end.
Thanks

**Comment by dikshachauhan-qasource** [comment]
Hi @AndersonQ 

We have revalidated all the artifacts shared with us on 8.1 kibana build and found that:

- With only system integration all the agents were shipping data to Kibana and were in healthy state, with filebeat and metricbeat binaries in running state. Test using :
     - .tar artifact on Mac 12
     - universal artifact on Mac 12
     - universal artifact on M1
     - aarch artifact on M1
- However, when endpoint security is added to agent it goes to unhealthy state.

**Screenshot:**
image

logs for M1 agent with aarch artifact : [elastic-agent-diagnostics-2022-03-10T10-40-48Z-00.zip](https://github.com/elastic/beats/files/8222776/elastic-agent-diagnostics-2022-03-10T10-40-48Z-00.zip)
[error-logsfor M1withUniversal-artifact.txt](https://github.com/elastic/beats/files/8222781/error-logsfor.M1withUniversal-artifact.txt)

Build details:
BUILD: 50485

Thanks
QAS

**Comment by jasonrhodes** [comment]
@AndersonQ I see that @elastic/infra-monitoring-ui is added a codeowner reviewer here, but I don't see any files owned by us there. Can you help me understand why that might be the case? (Did it at one point include changes to the kibana, elasticsearch, logstash, or beats modules, for example?)

File: .ci/packaging.groovy
Comment (belimawr) [inline .ci/packaging.groovy:215]: What is the plan here? Comment it in once CI is green and merge or do it in another PR?

**Review by belimawr — COMMENTED** [review]
LGTM, but I have a couple of questions.

**Review by ph — APPROVED** [review]
I think @belimawr comment is good concerning the mage dependencies to ensure it work.
Other than that, I've tested this the associated beats branch and I was able to get the artifacts built.  Correctly

File: auditbeat/magefile.go
Comment (AndersonQ) [reply auditbeat/magefile.go:80]: [Mage dependencies](https://v1.magefile.org/dependencies/) will ensure a function was called, here I need more than just that. I need the build to have been invoked with the right environment variables, it has happened 2 times (once for each architecture). Thus it isn't really a solution here.

File: .ci/packaging.groovy
Comment (AndersonQ) [reply .ci/packaging.groovy:215]: I ended up testing it on the elastic-agent repo, didn't do what I needed. Besides in order to unblock the release team, better to merge it now and adjust the CI later. But good catch anyway :)

**Pull request merged**
